File: src/main/java/com/beat/global/auth/client/kakao/response/KakaoAccessTokenResponse.java
package com.beat.global.auth.client.kakao.response;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public record KakaoAccessTokenResponse(
	@JsonProperty("token_type")
	String tokenType,

	@JsonProperty("access_token")
	String accessToken,

	@JsonProperty("refresh_token")
	String refreshToken,

	@JsonProperty("expires_in")
	Long expiresIn,

	@JsonProperty("refresh_token_expires_in")
	Long refreshTokenExpiresIn
) {
}

File: infrastructure/src/main/kotlin/com/beat/infrastructure/external/social/kakao/response/KakaoAccessTokenResponse.kt
package com.beat.infrastructure.external.social.kakao.response

import com.fasterxml.jackson.annotation.JsonProperty

internal data class KakaoAccessTokenResponse(
    @param:JsonProperty("token_type")
    val tokenType: String?,
    @param:JsonProperty("access_token")
    val accessToken: String?,
    @param:JsonProperty("expires_in")
    val expiresIn: Int?,
    @param:JsonProperty("refresh_token")
    val refreshToken: String?,
    @param:JsonProperty("refresh_token_expires_in")
    val refreshTokenExpiresIn: Int?,
)
